### DISCUSSION TITLE
1D Euler-like equation steady state: unwanted oscillation

### DISCUSSION THREAD

**Discussion by ethrelfall** [firedrakeproject/firedrake#3094] [Firedrake support]
Dear community,

I'm trying to solve a simple example of a source problem in 1D ([-1,1]), constant source n* everywhere, sonic outflow at boundaries.

![image](https://github.com/firedrakeproject/firedrake/assets/77395015/d7bd2198-2bcf-4722-8477-fdde59b07ea0)

There is simple analytic solution:

![image](https://github.com/firedrakeproject/firedrake/assets/77395015/487acab9-69b3-4182-b266-c81cd1467eb4)

My question is why does Firedrake give oscillatory-looking solutions (grid-scale wiggle).  It goes away as the mesh is refined but it looks to me like something just isn't right:

![image](https://github.com/firedrakeproject/firedrake/assets/77395015/23f0e513-2fff-469e-a7f3-9b46a2c6106b)

Script attached.
[SOL_1D_example.zip](https://github.com/firedrakeproject/firedrake/files/12565684/SOL_1D_example.zip)

**Comment by ethrelfall** [comment]
Remembered I asked a quite similar question recently so I'll check this isn't same problem, sorry ...

**Comment by colinjcotter** [comment]
It’s a hyperbolic problem so you need to use a method appropriate to that: SUPG or edge stabilisation for CG, or DG with a dissipative flux.

On 9 Sep 2023, at 13:18, ethrelfall ***@***.***> wrote:

﻿

Remembered I asked a quite similar question recently so I'll check this isn't same problem, sorry ...

**Comment by colinjcotter** [comment]
This book explains some of the issues in the continuous finite element case.

<https://onlinelibrary.wiley.com/doi/book/10.1002/0470013826>
Finite Element Methods for Flow Problems<https://onlinelibrary.wiley.com/doi/book/10.1002/0470013826>
onlinelibrary.wiley.com<https://onlinelibrary.wiley.com/doi/book/10.1002/0470013826>
[0470013826.gif]<https://onlinelibrary.wiley.com/doi/book/10.1002/0470013826>


On 9 Sep 2023, at 13:29, Cotter, Colin J ***@***.***> wrote:

﻿
It’s a hyperbolic problem so you need to use a method appropriate to that: SUPG or edge stabilisation for CG, or DG with a dissipative flux.

On 9 Sep 2023, at 13:18, ethrelfall ***@***.***> wrote:

﻿

Remembered I asked a quite similar question recently so I'll check this isn't same problem, sorry ...

**Comment by ethrelfall** [comment]
Hi Colin, many thanks for the answer and the reference.

I have to admit I don't understand why you refer to this as an hyperbolic system when it's just an ODE - ?

(My understanding is that hyperbolic eqs are stuff like dn/dt = dn/dx, or wave equation, or Maxwell's equations.)

**Comment by colinjcotter** [comment]
It’s anything that can be solved by integration along a characteristic line. In your equation, that line is just the coordinate. It means that the solution only depends on values upstream of that line.

Contrast with u_{xx} = f(x),
which cannot be solved that way (not in one step, anyway). The solution depends on values everywhere.



On 9 Sep 2023, at 20:29, ethrelfall ***@***.***> wrote:

﻿

Hi Colin, many thanks for the answer and the reference.

I have to admit I don't understand why you refer to this as an hyperbolic system when it's just an ODE - ?

(My understanding is that hyperbolic eqs are stuff like dn/dt = dn/dx, or wave equation, or Maxwell's equations.)

**Comment by colinjcotter** [comment]
When you solve an ODE numerically the usual way, starting from an initial condition, you are making use of this structure naturally. When you solve for all values of time at once, you have to use a discretisation that is stable in that framework.

On 10 Sep 2023, at 09:36, Cotter, Colin J ***@***.***> wrote:

﻿
It’s anything that can be solved by integration along a characteristic line. In your equation, that line is just the coordinate. It means that the solution only depends on values upstream of that line.

Contrast with u_{xx} = f(x),
which cannot be solved that way (not in one step, anyway). The solution depends on values everywhere.



On 9 Sep 2023, at 20:29, ethrelfall ***@***.***> wrote:

﻿

Hi Colin, many thanks for the answer and the reference.

I have to admit I don't understand why you refer to this as an hyperbolic system when it's just an ODE - ?

(My understanding is that hyperbolic eqs are stuff like dn/dt = dn/dx, or wave equation, or Maxwell's equations.)
